api/views/v1/projects.py

```diff
@@ -164,34 +171,58 @@ def delete(self, request):
             });
         """
 
-        responseData, httpStatus = self.deleteProjects(request.query_params)
+        responseData, httpStatus = self.deleteProjects(
+            request.query_params, getAuthorizationToken(request)
+        )
         return Response(responseData, status=httpStatus)
 
     @staticmethod
-    def getProjects(projectData):
+    def getProjects(projectData, authorizationToken):
         """
         Service API function that can be called internally as well as through the API to get
         project data based on input data.
 
-        @param projectData      Dict for a single project dict or list of dicts for multiple tasks.
+        @param projectData      Dict for a single project.
+        @param authorizationToken      JWT authorization token.
         @return      A tuple of (response_data, http_status).
         """
         try:
 
-            projects = Project.objects.filter(**projectData)  # Query mongo
-            serializer = ProjectSerializer(projects, many=True)
+            # Only get project where user is a contributor
+            userID = decodeApiKey(authorizationToken).get("userID")
 
+            if "userIDs" not in projectData:
+                projectData["userIDs"] = []
+            if not isinstance(projectData["userIDs"], list):
+                projectData["userIDs"] = [projectData["userIDs"]]
+            if userID not in projectData["userIDs"]:
+                projectData["userIDs"].append(userID)
+
+            userIDs = projectData["userIDs"]
+            del projectData["userIDs"]
+
+            projects = Project.objects.filter(
+                **projectData, userIDs__all=userIDs
+            )  # Query mongo
```

Comment:
```suggestion
            projects = Project.objects.filter(
                **projectData, userIDs__all=projectData["userIDs"]
            )  # Query mongo
```

Reply:
Lines 201-206 should not be deleted because than projectData["userIDs"] will remain in the `**projectData` as well as `userIDs__all=projectData["userIDs"]` part of the query which will mess the query up.

Reply:
alright, well in that case, you should be able to use `.pop()` do the same things

```suggestion
            projects = Project.objects.filter(
                **projectData, userIDs__all=projectData.pop("userIDs")
            )  # Query mongo
```

Reply:
This is a cool solution, I was a bit worried about the concurrency of unpacking projectData and popping it at the same time, but I think it will work. I'm going to flip the order of the params for clarity, though.